File: 56-merge-intervals/56. Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        stack = []
        intervals.sort()
        for start, end in intervals:
            if len(stack) == 0:
                stack.append([start, end])
                continue
            if stack[-1][1] >= start:
                prevStart, prevEnd = stack.pop()
                stack.append([min(prevStart, start), max(prevEnd, end)])
            else:
                stack.append([start, end])
        return stack

File: 56-merge-intervals/56. Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> stack;
        for (auto& arr : intervals) {
            int s = arr[0], e = arr[1];
            if (stack.empty()) {
                stack.push_back({s, e});
                continue;
            }
            if (s <= stack[stack.size() - 1][1]) {
                stack[stack.size() - 1] = {stack[stack.size() - 1][0], max(stack[stack.size() - 1][1], e)};
            } else {
                stack.push_back({s, e});
            }
        }
        return stack;
    }
};